integrated_channels/sap_success_factors/models.py

```diff
@@ -93,9 +121,37 @@ class LearnerDataTransmissionAudit(models.Model):
     error_message = models.TextField(blank=True)
     created = models.DateTimeField(auto_now_add=True)
 
+    epoch = datetime(1970, 1, 1, tzinfo=timezone.utc)
+
     class Meta:
         app_label = 'sap_success_factors'
 
+    def __init__(self, *args, **kwargs):
+        """
+        Return a new ``LearnerDataTransmissionAudit`` object.
+
+        Optional arguments:
+        * ``enterprise_enrollment``: If provided, then ``enterprise_course_enrollment_id``, ``user_id``, * ``course_id``
+          are pulled from this EnterpriseCourseEnrollment object.
+        * ``certificate``: If provided, then ``course_completed=True``; ``completed_timestamp`, ``grade``
+          are pulled from this GeneratedCertificate object.
+
+        Otherwise, the field values are pulled directly from the kwargs, as usual.
+        """
+        enterprise_enrollment = kwargs.pop('enterprise_enrollment', None)
+        if enterprise_enrollment is not None:
+            kwargs['enterprise_course_enrollment_id'] = enterprise_enrollment.id
+            kwargs['sapsf_user_id'] = enterprise_enrollment.enterprise_customer_user.get_remote_id()
+            kwargs['course_id'] = enterprise_enrollment.course_id
+
+        certificate = kwargs.pop('certificate', None)
+        if certificate is not None:
+            kwargs['course_completed'] = True
+            kwargs['completed_timestamp'] = int((certificate.created_date - self.epoch).total_seconds())
```

Comment:
The learner completion documentation is inconclusive.  The example given doesn't provide a number:

```
{
"userID": "<USERID>",
"courseID":"<COURSEID>",
"providerID":<PROVIDERID>,
"courseCompleted":"true",
"completedTimestamp":<COMPLETEDTIMESTAMP>,
"instructorName":"<InstructorName>",
"grade":"<grade>",
"totalHours":"<TOTALHOURS>",
"creditHours":<CREDITHOURS>,
"contactHours":<CONTACTHOURS>,
"cpeHours":<CPE>,
"comments":<COMMENTS>,
"price":<PRICE>,
"currency":<CURRENCY>
}
```

And the field description doesn't help: 
> This is the Unix Epoch time that the user completed the course. SAP SuccessFac­tors Learning uses this completion date for the audit trail of users' completion.